src/modules/interactions/buttons/settings.js

```diff
@@ -0,0 +1,97 @@
+import EmbedPageMessage from '../../../dto/EmbedPageMessage';
+import { getMember, getRole, userDb } from '../../../util';
+
+const command = 'settings';
+
+/**
+ *
+ * @param {import('discord.js').GuildMember} member
+ * @param {EmbedPage} page
+ * @param {String} roleName
+ * @returns {Promise<void>}
+ */
+async function handleToggle(member, page, roleName) {
+    const m = await member.fetch(true);
+    const role = await getRole(roleName);
+    if (role === undefined) {
+        const userDto = userDb.get(member.id);
+        if (!userDto.notifications.get(roleName)) {
+            userDto.notifications.set(roleName, true);
+            page.buttons[0].setLabel('unsubscribe').setStyle('DANGER');
+        } else {
+            userDto.notifications.set(roleName, false);
+            page.buttons[0].setLabel('subscribe').setStyle('SUCCESS');
+        }
+    } else if (m.roles.cache.has(role)) {
+        await m.roles.remove(role);
+        page.buttons[0].setLabel('subscribe').setStyle('SUCCESS');
+    } else {
+        await m.roles.add(role);
+        page.buttons[0].setLabel('unsubscribe').setStyle('DANGER');
+    }
+}
+
+/**
+ *
+ * @param {import('discord.js').GuildMember} member
+ * @param {EmbedPage} page
+ * @param {String} roleName
+ * @returns {Promise<void>}
+ */
+async function handleOS(member, page, roleName) {
+    const osToIndexMap = {
+        windows: 0,
+        linux: 1,
+        mac: 2,
+    };
+    const buttonIndex = osToIndexMap[roleName];
+
+    const role = await getRole(roleName);
+
+    if (member.roles.cache.has(role)) {
+        await member.roles.remove(role);
+        page.buttons[buttonIndex].setStyle('SUCCESS');
+    } else {
+        await member.roles.add(role);
+        page.buttons[buttonIndex].setStyle('DANGER');
+    }
+}
+
+/**
+ *
+ * @param {import('discord.js').Client} client
+ * @param {import('discord.js').ButtonInteraction} interaction
+ * @param {string[]} args
+ */
+async function fun(client, interaction, args) {
+    const page = EmbedPageMessage.getInstance(interaction.message.id);
+    if (!page) {
+        await interaction.update({
+            content: 'This settings has expired. Please run `settings` again!',
```

Comment:
Is this necessary? Could we not maybe make a new page to continue on from.

Reply:
not anymore, as pages have a timeout of 3 minutes now